examples/porting/lv_port_lcd_stm32_template.c

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file lv_port_lcd_stm32_template.c
+ *
+ * Example implementation of the LVGL LCD display drivers on the STM32 platform
+ */
+
+/*Copy this file as "lv_port_disp.c" and set this value to "1" to enable content*/
+#if 0
+
+/*********************
+ *      INCLUDES
+ *********************/
+/* Include STM32Cube files here, e.g.:
+#include "stm32f7xx_hal.h"
+*/
+
+#include "lv_port_disp.h"
+#include "./src/drivers/display/st7789/lv_st7789.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+#ifndef MY_DISP_HOR_RES
+    #warning Please define or replace the macro MY_DISP_HOR_RES with the actual screen width, default value 320 is used for now.
+    #define MY_DISP_HOR_RES    320
+#endif
+
+#ifndef MY_DISP_VER_RES
+    #warning Please define or replace the macro MY_DISP_VER_RES with the actual screen height, default value 240 is used for now.
+    #define MY_DISP_VER_RES    240
+#endif
+
+#define BUS_SPI1_POLL_TIMEOUT 0x1000U
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static void lcd_color_transfer_ready_cb(SPI_HandleTypeDef * hspi);
+static int32_t lcd_io_init(void);
+static void lcd_send_cmd(lv_display_t * disp, const uint8_t * cmd, size_t cmd_size, const uint8_t * param,
+                         size_t param_size);
+static void lcd_send_color(lv_display_t * disp, const uint8_t * cmd, size_t cmd_size, uint8_t * param,
+                           size_t param_size);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static lv_display_t * lcd_disp;
+static volatile int lcd_bus_busy = 0;
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_port_display_init(void)
+{
+    /* Initialize LCD I/O */
+    if(lcd_io_init() != 0)
+        return;
+
+    /* Create the LVGL display object and the ST7789 LCD display driver */
+    lcd_disp = lv_st7789_create(MY_DISP_HOR_RES, MY_DISP_VER_RES, LV_LCD_FLAG_NONE, lcd_send_cmd, lcd_send_color);
+    lv_display_set_rotation(lcd_disp, LV_DISPLAY_ROTATION_270);     /* set landscape orientation */
+
+    /* Example: two dynamically allocated buffers for partial rendering */
+    lv_color_t * buf1 = NULL;
+    lv_color_t * buf2 = NULL;
+
+    uint32_t buf_size = MY_DISP_HOR_RES * MY_DISP_VER_RES / 10 * lv_color_format_get_size(lv_display_get_color_format(
+                                                                                              lcd_disp));
+
+    buf1 = lv_malloc(buf_size);
```

Comment:
Use global or static variables instead of malloc.

Reply:
OK, I'll change it.